### PR TITLE
Release 4.3.0

### PR DESCRIPTION
**Pull request by maminrayej**
[bot] Release wasmer version 4.3.0

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,6 +9,32 @@ Looking for changes that affect our C API? See the [C API Changelog](lib/c-api/C
 
 ## **Unreleased**
 
+## 4.3.0 - 10/05/2024
```
Comment (Arshia001) [inline CHANGELOG.md:12]: Summary of changes please!

**Review by Arshia001 — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (maminrayej) [reply CHANGELOG.md:12]: I disabled auto-merge, it's fine. I'll enable it after I added the manual commits.
We first update `0.x` crates last time and then update the change log.
I'll do it in that order.

Comment (Arshia001) [reply CHANGELOG.md:12]: The order doesn't really matter. CI will need to rerun after you push the changes, so push it all in one go

Comment (maminrayej) [reply CHANGELOG.md:12]: I prefer to do it in separate steps for now. I'll do it in one go when I'm more confident.

**Pull request merged**